eda-env/Lib/site-packages/parso/cache.py

```
@@ -0,0 +1,275 @@
+        pickling = False
+
+    item = _NodeCacheItem(module, lines, p_time)
+    _set_cache_item(hashed_grammar, path, item)
+    if pickling and path is not None:
+        try:
+            _save_to_file_system(hashed_grammar, path, item, cache_path=cache_path)
+        except PermissionError:
+            # It's not really a big issue if the cache cannot be saved to the
+            # file system. It's still in RAM in that case. However we should
+            # still warn the user that this is happening.
+            warnings.warn(
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)